cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -172,8 +172,9 @@ private Resource calculateShapeBasedOnResource(@Nonnull Configuration configurat
 
     String namespacePrefix = shacl.getNsURIPrefix(originalResource.getNameSpace());
     if (namespacePrefix == null) {
-      throw new RuntimeException("no name alternative found for '" + originalResource.getURI() + "':"
-                                 + " please add namespace to prefixes.");
+      long nrTempPrefixes = shacl.getNsPrefixMap().keySet().stream().filter(key -> key.startsWith("ns")).count();
```

Comment:
hmm, no I think about it, this ok in most cases but what if there was already one or more "ns" prefixes present. the safer option is to find a good starting point and go from there.

roughly like 

```
private int getAvailableNamespaceIndex(...) {
  return shacl.getNsPrefixMap().keySet().stream()
            .filter(key -> key starts with)
            .map(ns -> ns.substring(2))
            .mapToInt(ns -> Integer.toInt(ns))
            .max();
} 
```